staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -181,17 +182,23 @@ func (d *CachedDiscoveryClient) writeCachedFile(filename string, obj runtime.Obj
 		return err
 	}
 
-	f, err := ioutil.TempFile(filepath.Dir(filename), filepath.Base(filename)+".")
+	// Use os.Create to get default file permissions honoring umask
+	f, err := os.Create(filename + "." + string(uuid.NewUUID()))
 	if err != nil {
 		return err
 	}
 	defer os.Remove(f.Name())
-	_, err = f.Write(bytes)
+
+	// remove world-wide permissions before writing data to the file
+	info, err := os.Stat(f.Name())
 	if err != nil {
 		return err
 	}
-
-	err = os.Chmod(f.Name(), 0660)
+	err = os.Chmod(f.Name(), info.Mode().Perm()&0660)
```

Comment:
os.Create() will create file with mode 0666, here `info.Mode().Perm()&0660` changes the mode to `0660`.

So, seems nothing changes. 

what're the benefits?

Reply:
os.Create honors umask, so will at most create a file with 0666. Most non root users default to umask of 0022, which means os.Create produces files of 0644 for them. We drop the world permissions and they end up with 0640.

Reply:
Ok, I understand. 

That means the final permission will depend on the user umask, we just drop the world permission as a supplement.

A novel solution I think, just a little complicated. :)

Should we keep group-writeable for files in this case? Can you take an example?